### DISCUSSION TITLE
What causes this theming quirk?

### DISCUSSION THREAD

**Discussion by kellan-dev** [vercel/next.js#74092] [Help]
### Summary

I stumbled on this weird quirk the other day by accident when I was setting up dark mode with next-themes. When switching to dark mode with my dropdown menu, the "dark" class was being set correctly on the html element, but the color variables defined in the ".dark" ruleset weren't being applied. 

It took a while to narrow down, but what I found is that there seems to be some kind of lower level dependency on the word "dark" (all lowercase) being present somewhere in the code. It can be the name of a function, the name of a variable, the value of a variable or argument, or even in a comment.  Without it, theming breaks in the way I described above. But having "dark" somewhere in the code causes it to work properly again. I tried other terms and variations like "light" and "system" and "darkmode", but they had no effect. So it seems to be the term "dark" specifically that matters.

I would be interested to know what might cause this kind of behavior to occur, if anyone has any insight. 

### Example

https://github.com/kellan-dev/themes-bug

**Comment by icyJoseph** [comment]
🤔 perhaps it comes from tailwind preprocessing?

> If you're using the value prop to specify different attribute values, make sure your dark theme explicitly uses the "dark" value, as required by Tailwind.

Funny enough, if a className has `dark:` then it doesn't work, but adding that as a comment, or value, seemed to work.

Maybe take that repository to tailwind and ask over there? I am not that familiar with that codebase, so I am not sure where they do the file walk, and what checks they do when visiting each file.

Stackblitz fork that showcases the issue: https://stackblitz.com/edit/github-qc1exkvg?file=components%2Fdark.ts,tailwind.config.ts,app%2Flayout.tsx,components%2Ftheme-toggle.tsx,components%2Fui%2Fbutton.tsx,app%2Fpage.tsx
